### ISSUE TITLE
addGeoJSON issue

### ISSUE THREAD

**Issue by Virastrum** [rstudio/leaflet#146]
I have been unable to get addGeoJSON to work when placed in a shiny application. The code I'm using works perfectly outside of a shiny application. addMarkers is functional in shiny but the addGeoJSON code seem to be ignored. The code I'm using is below

`map_layer_counties <-leaflet() %>%
      addTiles() %>%
      setMaxBounds(-12, 50, -4, 57) %>%
      addGeoJSON(ireland_json)`

This is correctly generating the map in the shiny ui but without isn't including any of the GeoJSON information. It isn't throwing any sort of error. ireland_json is correctly formatted and exists.

Thanks for any help you can give me.

**Comment by jcheng5** [comment]
Is there an error in your web browser's JavaScript error console?

Can you tell us the exact contents of ireland_json (maybe in a gist)?

**Comment by Virastrum** [comment]
There isn't any error in the javascript console. However, nonsensical items do exist in the `<svg class ="leaflet-zoom-animated"> ` tag. These items do not appear on the map and their locations change depending on the zoom level of the map (at certain zoom levels they are in Greenland, other times in Ireland and other times not even on the leaflet map but on other parts of the shiny application) It's quite confusing as the JSON file and code works perfectly when not placed in a Shiny application.

You can find the JSON file I am using here: http://gist.github.com/aerrity/4338818

**Comment by cholcomb** [comment]
Did you use `ireland_json <- fromJSON("ireland_json.txt",simplifyVector = FALSE)`

**Comment by Virastrum** [comment]
No, instead I used

`ireland_json <- readLines("Location\Ireland.json") %>% paste(collapse = "\n")`

This has worked fine outside of the Shiny application for reading in the JSON file and outputting it on a Leaflets map.

**Comment by jcheng5** [comment]
Would it be possible for you to give us a reproducible example? Or at least the data?

**Comment by cholcomb** [comment]
FWIW, I tried it with the `ireland.json` file inside the gist in Shiny with both input functions and I don't see any issues. It's probably something else not shared.

**Comment by Virastrum** [comment]
Thanks for all your help with this. I have isolated the issue to a conflict between Leaflets and RCharts when using the NVD3 JavaScript libraries.

For anyone else in the future who finds this issue I will post the solution I discovered. Commenting out the following code block in the nv.d3.css file of rCharts will prevent the conflict between rCharts and Leaflet.

`svg {
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  /* Trying to get SVG to act like a greedy block in all browsers */
  display: block;
  width:100%;
  height:100%;
}`
